browser/ui/views/tabs/shared_pinned_tab_dummy_view_views.cc

```diff
@@ -0,0 +1,140 @@
+                    .SetBorder(views::CreateThemedRoundedRectBorder(
+                        kThumbnailBorderThickness, kThumbnailRadius,
+                        kColorBraveSharedPinnedTabDummyViewThumbnailBorder))
+                    .SetPreferredSize(kThumbnailSize)
+                    .SetLayoutManager(std::make_unique<views::FillLayout>())
+                    .AddChild(views::Builder<views::ImageView>()
+                                  .CopyAddressTo(&thumbnail_view_)
+                                  .SetImageSize(kThumbnailImageSize)))
+      .AddChild(views::Builder<views::Label>()
+                    .CopyAddressTo(&title_label_)
+                    .SetText(l10n_util::GetStringUTF16(
+                        IDS_SHARED_PINNED_TABS_DUMMY_TAB_VIEW_TITLE))
+                    .SetFontList(
+                        gfx::FontList()
+                            .DeriveWithSizeDelta(kTitleFontSize -
+                                                 gfx::FontList().GetFontSize())
+                            .DeriveWithWeight(gfx::Font::Weight::SEMIBOLD))
+                    .SetEnabledColorId(kColorBraveSharedPinnedTabDummyViewTitle)
+                    .SetProperty(views::kMarginsKey, gfx::Insets().set_top(40)))
+      .AddChild(
+          views::Builder<views::Label>()
+              .CopyAddressTo(&description_label_)
+              .SetText(l10n_util::GetStringUTF16(
+                  IDS_SHARED_PINNED_TABS_DUMMY_TAB_VIEW_DESCRIPTION))
+              .SetFontList(gfx::FontList().DeriveWithSizeDelta(
+                  kDescriptionFontSize - gfx::FontList().GetFontSize()))
+              .SetEnabledColorId(kColorBraveSharedPinnedTabDummyViewDescription)
+              .SetProperty(views::kMarginsKey, gfx::Insets().set_top(8)))
+      .BuildChildren();
+
+  SkPath path;
+  path.addRoundRect(
+      SkRect::MakeWH(kThumbnailImageSize.width(), kThumbnailImageSize.height()),
+      kThumbnailRadius - kThumbnailBorderThickness,
+      kThumbnailRadius - kThumbnailBorderThickness);
+  thumbnail_view_->SetClipPath(path);
+
+  subscription_->SetSizeHint(kThumbnailImageSize);
+  subscription_->SetUncompressedImageCallback(base::BindRepeating(
+      [](views::ImageView* thumbnail_view, gfx::ImageSkia image) {
+        thumbnail_view->SetImage(ui::ImageModel::FromImageSkia(image));
+      },
+      thumbnail_view_));
+  thumbnail_->RequestThumbnailImage();
+}
+
+SharedPinnedTabDummyViewViews::~SharedPinnedTabDummyViewViews() = default;
+
+void SharedPinnedTabDummyViewViews::Install() {
+  auto* browser = chrome::FindBrowserWithTab(dummy_contents_);
+  CHECK(browser);
+
+  // Borrows WebView::SetCrashedOverlayView() which is used to attach SadTabView
+  // over views::WebView.
+  // TODO(sko) We should take split view into account. This is the same problem
+  // as with SadTabView.
+  static_cast<BrowserView*>(browser->window())
+      ->contents_web_view()
+      ->SetCrashedOverlayView(this);
```

Comment:
I guess the DummyContents can't really crash, but does this replace the crashed view (so if it did crash, we won't show anything).

Probably fine, as its always empty, but just curious

Reply:
Good question. As we suppress renderer process for dummy contents view, I believe it won't happen.

https://sourcegraph.com/github.com/brave/brave-core@7c164e9ebce9d000848bd17f35422a685f58d779/-/blob/browser/ui/tabs/shared_pinned_tab_service.cc?L824

But if the crashed view were attached before this,  yes, it'd be replaced.